### PR TITLE
chore: Simplifies makefile

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Simplifies makefile now that there are not special folders like integrationtesting.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: GNUmakefile [hunk 1]
```diff
@@ -35,17 +31,17 @@ install: fmtcheck
 
 .PHONY: test
 test: fmtcheck
-	go test $(TEST) -timeout=30s -parallel=4 -race -covermode=atomic -coverprofile=coverage.out
+	go test ./... -timeout=30s -parallel=4 -race
```
Comment (AgustinBettati) [inline GNUmakefile:34]: just for reference I see this was included originally for in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1496, but we no longer include these converage reports as PR comments.

File: .github/workflows/acceptance-tests-runner.yml [hunk 2]
```diff
@@ -96,7 +96,6 @@ on:
         required: true
     
 env:
-  TF_ACC: 1
```
Comment (AgustinBettati) [inline .github/workflows/acceptance-tests-runner.yml]: while redundant I do see the value in leaving the TF_ACC env var just for clarity

**Review by AgustinBettati — APPROVED** [review]
LGTM

File: GNUmakefile (hunk 1 above)
Comment (lantoli) [reply GNUmakefile:34]: yes, at the moment we don't include any coverage report

File: .github/workflows/acceptance-tests-runner.yml (hunk 2 above)
Comment (lantoli) [reply .github/workflows/acceptance-tests-runner.yml]: don't know, if we assume "make testacc" does it i would remove it from here.

i would choose only one place to have it

Comment (oarbusi) [reply .github/workflows/acceptance-tests-runner.yml]: [q] Does this mean that TF_ACC is not needed to run acceptance tests? Including locally? If so, we should update Terraform Development Guide

File: GNUmakefile [hunk 3]
```diff
@@ -81,13 +77,6 @@ tools:  ## Install dev tools
 	go install github.com/hashicorp/terraform-plugin-codegen-framework/cmd/tfplugingen-framework@latest
 	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(shell go env GOPATH)/bin $(GOLANGCI_VERSION)
 
-.PHONY: check
-check: test lint
```
Comment (andreaangiolillo) [inline GNUmakefile:85]: why do we remove this? I would include additional check to this command such as fmtcheck

File: .github/workflows/acceptance-tests-runner.yml (hunk 2 above)
Comment (andreaangiolillo) [reply .github/workflows/acceptance-tests-runner.yml]: +1 I see the value in keeping it

File: GNUmakefile (hunk 3 above)
Comment (lantoli) [reply GNUmakefile:85]: it's not being used in GH actions or commit prechecks and i don't think anybody is using this

File: .github/workflows/acceptance-tests-runner.yml (hunk 2 above)
Comment (lantoli) [reply .github/workflows/acceptance-tests-runner.yml]: @oarbusi  testacc makefile goal is: TF_ACC=1 go test ...

File: GNUmakefile (hunk 3 above)
Comment (andreaangiolillo) [reply GNUmakefile:85]: ah okay, I was probably the only one using it xD happy to remove it then

File: .github/workflows/acceptance-tests-runner.yml (hunk 2 above)
Comment (lantoli) [reply .github/workflows/acceptance-tests-runner.yml]: change reverted, i'll keep it, thx for the feedback

**Comment by lantoli** [comment]
changing only makefile

**Pull request merged**